packages/key-management/test/KeyAgentBase.test.ts

```diff
@@ -64,11 +71,66 @@ describe('KeyAgentBase', () => {
     // creates a new array obj
     expect(keyAgent.knownAddresses).not.toBe(initialAddresses);
 
-    const sameAddress = await keyAgent.deriveAddress({ index, type });
+    const sameAddress = await keyAgent.deriveAddress({ index, type }, 0);
     expect(sameAddress.address).toEqual(address.address);
     expect(keyAgent.knownAddresses.length).toEqual(1);
   });
 
+  test('deriveAddress derives the address with stake key at index 0 if no stake key index is provided', async () => {
```

Comment:
I think this test can be removed. The description is not appropriate anymore and I think it's covered by the next test